packages/devtools_app/lib/src/debugger/flutter/debugger_model.dart

```diff
@@ -6,13 +6,48 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart' hide Stack;
 import 'package:vm_service/vm_service.dart';
 
+/// A tuple of a script and an optional location.
+class ScriptLocation {
+  ScriptLocation(this.scriptRef, {this.location});
+
+  final ScriptRef scriptRef;
+
+  /// This field can be nullable.
+  final SourcePosition location;
+
+  @override
+  bool operator ==(other) {
+    return other is ScriptLocation &&
+        other.scriptRef == scriptRef &&
+        other.location == location;
+  }
+
+  @override
+  int get hashCode => (scriptRef.hashCode << 7) ^ (location?.hashCode ?? 0);
```

Comment:
this is a fine hashcode but might want to use the flutter helper method for generating a hashcode of a couple fields just to keep people on the right track if copying this code.

Reply:
Ah, I didn't know about the `hashValues()` util - will change to use it.